src/lib.rs

```diff
@@ -220,27 +220,28 @@ pub fn load_metadata(
 
     let metadata = command.exec().context("failed to load cargo metadata")?;
 
-    for package in &metadata.packages {
-        for dep in &package.dependencies {
-            if dep.rename.as_deref().unwrap_or(dep.name.as_str()) != BINDINGS_CRATE_NAME {
-                continue;
-            }
+    if !ignore_version_mismatch {
+        for package in &metadata.packages {
+            for dep in &package.dependencies {
+                if dep.rename.as_deref().unwrap_or(dep.name.as_str()) != BINDINGS_CRATE_NAME {
+                    continue;
+                }
 
-            if ignore_version_mismatch
-                || dep
-                    .req
-                    .matches(&Version::parse(env!("CARGO_PKG_VERSION")).unwrap())
-            {
-                continue;
-            }
+                let req = VersionReq::parse(env!("CARGO_PKG_VERSION")).unwrap();
+                let v = dep.req.to_string();
+                match v.strip_prefix('^').unwrap_or(&v).parse::<Version>() {
+                    Ok(v) if req.matches(&v) => break,
```

Comment:
Does this work if the patch version difference is the other way around, e.g. if `cargo-component` was bumped to 0.4.2?

Reply:
You're right, this isn't quite adequate. What do you think of this approach instead?

* If major/minor differ:
  * If `cargo-component` version is _newer_ than the bindings crate, function like we do now (emit warning, allow upgrade of `Cargo.toml`)
  * If `cargo-component` version is _older_ than the bindings crate, emit an error saying cargo-component must be upgraded and do not allow upgrades to `Cargo.toml`.

Reply:
That sounds like it would do the trick!

Reply:
Okay, I've pushed up a commit to hopefully address this and provide better feedback to users.